mlir/unittests/Tools/lsp-server-support/Transport.cpp

```diff
@@ -0,0 +1,121 @@
+//===- Transport.cpp - LSP JSON transport unit tests ----------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Tools/lsp-server-support/Transport.h"
+#include "mlir/Tools/lsp-server-support/Logging.h"
+#include "mlir/Tools/lsp-server-support/Protocol.h"
+#include "llvm/Support/FileSystem.h"
+#include "gmock/gmock.h"
+#include "gtest/gtest.h"
+
+using namespace mlir;
+using namespace mlir::lsp;
+using namespace testing;
+
+namespace {
+
+TEST(TransportTest, SendReply) {
```

Comment:
it seems that the stacked PR are not working properly :(

Reply:
Yes, unfortunately I can't submit PRs to LLVM from my fork, specifying a destination branch that only exists in my fork. As a result, this one PR contains the commit I want reviewed, as well as the commits stacked underneath it... I will rebase now that I've landed #89855, and that will cause the patch view to only display the top commit.